HerPortal/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace HerPortal.Middleware;
+
+public class SecurityHeadersMiddleware
+{
+    private readonly RequestDelegate next;
+
+    public SecurityHeadersMiddleware(RequestDelegate next)
+    {
+        this.next = next;
+    }
+
+    public Task Invoke(HttpContext context)
+    {
+        if (!context.Response.Headers.ContainsKey("X-Content-Type-Options"))
+        {
+            context.Response.Headers.Add("X-Content-Type-Options", "nosniff");
+        }
+        
+        if (!context.Response.Headers.ContainsKey("Content-Security-Policy"))
+        {
+            context.Response.Headers.Add("Content-Security-Policy",
+                "default-src 'self'; script-src * 'unsafe-inline' 'unsafe-eval'; connect-src  * 'unsafe-inline'"); 
```

Comment:
This is a pretty loose policy, and we probably could tighten it up (since we know exactly what content we're serving), but I'm not sure how worthwhile it is. If you have opinions, happy to look at it now, or we can wait and see what the pen-test results say... (but I'd be surprised if they didn't pick up on it)

The other thing that Kat raised is that this blocks inline scripts, which could break custom CSS overrides for accessibility (https://softwiretech.atlassian.net/browse/BEISHER-679) - given that we allow inline JS, it feels like we may as well allow inline CSS as well?

Reply:
Shall we wait until we make a decision about this in the other ticket to decide what this should be?

Reply:
Looks like we'll be asked to tighten this up in the pen test report